packages/go_router/lib/src/route.dart

```diff
@@ -447,8 +449,18 @@ class ShellRoute extends RouteBase {
   /// sub-route's builder.
   final ShellRoutePageBuilder? pageBuilder;
 
+  /// The observers for a shell route.
+  ///
+  /// The observers parameter is used by the [Navigator] built for this route.
+  /// sub-route's observers.
+  final List<NavigatorObserver>? observers;
+
   /// The [GlobalKey] to be used by the [Navigator] built for this route.
   /// All ShellRoutes build a Navigator by default. Child GoRoutes
   /// are placed onto this Navigator instead of the root Navigator.
   final GlobalKey<NavigatorState> navigatorKey;
+
+  /// HeroController for each nested Navigator, which solves cases where the
+  /// Hero Widget animation stops working when navigating.
+  final GoHero goHero = GoHero();
```

Comment:
I am not sure that it is the right concept to resolve `HeroController` issue. As for me, `GoRoute` is not guilty of knowing about `HeroController` at all.

@angjelkom Why not "create a cache in the RouteBuilder to store navkey to HeroController map" (as @chunhtai suggested)? It could be a simple `Map<GlobalKey<NavigatorState>, HeroController> _goHeroCache;` property in `RouteBuilder`. And when `HeroController` is needed, we check the cache and, if necessary, create the right one based on `isMaterialApp`/`isCupertinoApp`.